lib/labels.js

```diff
@@ -0,0 +1,254 @@
+  //   labelName <string> | <string[]>, optional, if not specified, all labels
+  //       will be deleted
+  // Returns: <Map>, labels that are already in use in issues/PRs
+  async deleteIn(repo, labelName) {
+    if (!labelName) {
+      labelName = (await this.getFrom(repo)).map(label => label.name);
+    }
+    if (!Array.isArray(labelName)) labelName = [labelName];
+
+    const usedLabels = await this.getUsedLabels(repo);
+    labelName = labelName.filter(name => !usedLabels.has(name));
+
+    await Promise.all(
+      labelName.map(name => this[request](this[getPath](repo, name), 'DELETE'))
+    );
+    return usedLabels;
+  }
+
+  // Delete label or labels in current repository
+  //   labelName <string> | <string[]>, optional, if not specified, all labels
+  //       will be deleted
+  async delete(labelName) {
+    return this.deleteIn(this.options.repo, labelName);
+  }
+
+  // Update label in `repoOwner/repoName`
+  // Either `labelName` or `label.name` should be specified.
+  // Signature: repo, label[, labelName]
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   label <Object>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>
+  //   labelName <string>, optional, if not
+  //       specified, `label.name` will be used instead
+  async updateIn(repo, label, labelName) {
+    if (!labelName) labelName = label.name;
+    if (!labelName) {
+      return Promise.reject(new Error('Label name was not specified'));
+    }
+    const { data } = await this[request](
+      this[getPath](repo, labelName),
+      'PATCH',
+      JSON.stringify(label)
+    );
+    return JSON.parse(data);
+  }
+
+  // Update label in current repository
+  // Either `labelName` or `label.name` should be specified.
+  // Signature: label[, labelName]
+  //   label <Object>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>
+  //   labelName <string>, optional, if not
+  //       specified, `label.name` will be used instead
+  async update(label, labelName) {
+    return this.updateIn(this.options.repo, label, labelName);
+  }
+
+  // Copy labels from `repoOwner/repoName` to current repository
+  //   labels <Object> | <Object[]> | <string> labels or repo
+  //   updateExistingLabels <boolean>
+  async copyFrom(labels, updateExistingLabels) {
+    if (typeof labels === 'string') labels = await this.getFrom(labels);
+    if (!Array.isArray(labels)) labels = [labels];
+    const curLabels = new Map(
+      (await this.get()).map(label => [label.name, label])
+    );
+
+    labels = await Promise.all(
+      labels.map(label => {
+        if (curLabels.has(label.name)) {
+          if (updateExistingLabels) return this.update(label);
+          return Promise.resolve(null);
+        } else {
+          return this.create(label);
+        }
+      })
+    );
+    return labels.filter(label => !!label);
```

Comment:
Pointless conversion:
```suggestion
    return labels.filter(label => label);
```